Algorithms/Recursion/Chess.java

```diff
@@ -0,0 +1,234 @@
+//Given an input N, a number. N represents the size of a chess board. We�ve a piece standing 
+//in top-left corner and it must reach the bottom-right corner. The piece moves as follows �
+//a. In any cell, the piece moves like a knight. But out of the possible 8 moves for a knight 
+//only the positive ones are valid i.e. both row and column must increase in a move.
+//b. On the walls (4 possible walls), the piece can move like a rook as well (in addition of 
+//knight moves). But, only the positive moves are allowed i.e. as a rook, piece can move any 
+//number of steps horizontally or vertically but in a manner, such that row or column must increase.
+//c. On the diagonals (2 possible diagonals), the piece can move like a bishop as well (in addition 
+//to the knight and possibly rook moves). But, only the positive moves are allowed i.e. as a bishop, 
+//piece can move in a way such that row and column must increase.
+
+//1. Write a recursive function which returns the count of different distinct ways this board can be 
+//crossed. Print the value returned.
+//
+// 2. Write a recursive function which returns an ArrayList of all valid paths. Print the value returned. 
+//E.g. Following are two valid paths for a board of size 4
+//{0-0}, {2-1}, {3-3}
+//{0-0}, {1-2}, {3-3}
+//
+//3. Write a recursive function which prints all valid paths.
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Chess {
+
+	public static void main(String[] args) {
+		// scanner
+		Scanner scn = new Scanner(System.in);
+
+		System.out.println("Enter size of board:");
+
+		// input for board size
+		int N = scn.nextInt();
+
+		System.out.println("Count of possible ways:");
+
+		// function call 1
+		System.out.print(countChessPaths(N, N, 0, 0));
+
+		System.out.println("Output of print path:");
+
+		// function call 2
+		printChessPaths(N, N, 0, 0, "");
```

Comment:
It seems like you have mixed **0-based indexing** with **1-based indexing**. As the size of the board is **NxN**, so if you are starting from *(0, 0)*, then take *(N - 1, N - 1)* as the end coordinate OR rather start from *(1, 1)*.